homeassistant/helpers/area_registry.py

```diff
@@ -210,8 +224,11 @@ def async_get_area(self, area_id: str) -> AreaEntry | None:
 
     @callback
     def async_get_area_by_name(self, name: str) -> AreaEntry | None:
-        """Get area by name."""
-        return self.areas.get_by_name(name)
+        """Get area by name or alias."""
+        if area := self.areas.get_by_name(name):
+            return area
+        areas_list = self.areas.get_areas_for_alias(name)
+        return areas_list[0] if areas_list else None
```

Comment:
This method is used by other methods in this class, like `async_create`, therefore you cannot change it here without consequences.

What you probably wanted to do is to create a separate method for your new purpose and call it in templates.

Reply:
> This method is used by other methods in this class, like `async_create`, therefore you cannot change it here without consequences.
> 
> What you probably wanted to do is to create a separate method for your new purpose and call it in templates.

That's what i suggested in Discord to devs, but we agreed to use same method.

Reply:
Who is `devs`? Who is `we`?

If there are some additional decisions made, they should be documented in the PR.

The code, as it is now, is buggy, and cannot be merged in this state.

Reply:
> Who is `devs`? Who is `we`?
> 
> If there are some additional decisions made, they should be documented in the PR.
> 
> The code, as it is now, is buggy, and cannot be merged in this state.

I posted link to Discord discussion.

As you may see in description, this approach was originally proposed by @synesthesiam . However, that PR was closed because it was using unoptimized `for`-loops. I used index array for it, but the logic remained the same.

Reply:
I see now that in the previous PR `balloob` had the same reservations about this approach as I do. This cannot be merged like that.

If you want to, we can discuss it further on Discord.

Reply:
> I see now that in the previous PR `balloob` had the same reservations about this approach as I do. This cannot be merged like that.
> 
> If you want to, we can discuss it further on Discord.

I'd love to! :) I actually missed that comment by Paulus!

Reply:
I wrote to you on Discord.